src/api/deposit/DepositApi.ts

```diff
@@ -67,7 +67,7 @@ export class DepositApiImpl implements DepositApi {
   public constructor(injectedDependencies: Params) {
     Object.assign(this, injectedDependencies)
 
-    this._contractPrototype = new this.web3.eth.Contract(batchExchangeAbi) as BatchExchangeContract
+    this._contractPrototype = (new this.web3.eth.Contract(batchExchangeAbi) as unknown) as BatchExchangeContract
```

Comment:
Do you know why this worked and now we require this?

Reply:
Previously our `ExtendedContract` could still be considered `web3.Contract` and could be cast to it
Because `web3.Contract.events is any`
But now typechain overrites not only something that was `any` but also `once` function + we override `getPastEvents`. So types become incompatible